src/main/java/be/syntra/devshop/DevshopFront/services/ProductServiceImpl.java

```diff
@@ -76,13 +77,21 @@ public StatusNotification addProduct(@Valid ProductDto productDto) {
     }
 
     @Override
-    public ProductList findAllProductsBySearchModel() {
-        ResponseEntity<ProductList> productListResponseEntity = restTemplate.postForEntity(resourceUrl + "/searching/", searchService.getSearchModel(), ProductList.class);
-        if (HttpStatus.OK.equals(productListResponseEntity.getStatusCode())) {
-            log.info("findAllProductsBySearchModel -> receivedFromBackEnd");
-            return productListResponseEntity.getBody();
+    public ProductListAndMinMaxPrice findAllProductsBySearchModel() {
+        log.info("findAllProductsBySearchModel() -> SearchModel -> {}", searchService.getSearchModel());
+        ResponseEntity<ProductListAndMinMaxPrice> productListResponseEntity = restTemplate.postForEntity(resourceUrl + "/searching/", searchService.getSearchModel(), ProductListAndMinMaxPrice.class);
+        if (productListResponseEntity.getBody().getProducts().size() == 0) {
```

Comment:
extract method 
```
if (productListIsEmpty(productListResponseEntity)){
...
}
```
-> 
```
private boolean productListIsEmpty(productListResponseEntity){
return productListResponseEntity.getBody().getProducts().size() == 0;
}
```

Reply:
dangerous check. You get a nullpointer if there's no list because of your getter. Instead -> 
Objects.requireNonNull(productListResponseEntity.getBody()).getProducts().isEmpty()
You can static import Objects, please.

Reply:
then again, further dissecting this method. You WANT the code to move through this check if you receive an empty list from the BE. So the check becomes: null == productListResponseEntity.getBody().getProducts()
Your second if statement is nested in an else, so that would become -> 
} else if (HttpStatus.OK.equals(productListResponseEntity.getStatusCode())) {
            log.info("findAllProductsBySearchModel() -> receivedFromBackEnd -> size : {}", productListResponseEntity.getBody().getProducts().size());
            return productListResponseEntity.getBody();
        }
And then, you're returning a productListResponseEntity.getBody(). Which could be null, if the first check fails. So it would be better to return a default new Object with desired parameters.

Reply:
total refactoring: 
@Override
    public ProductListAndMinMaxPrice findAllProductsBySearchModel() {
        ResponseEntity<ProductListAndMinMaxPrice> productListResponseEntity = restTemplate.postForEntity(resourceUrl + "/searching/", searchService.getSearchModel(), ProductListAndMinMaxPrice.class);
        if (HttpStatus.OK.equals(productListResponseEntity.getStatusCode())) {
            log.info("findAllProductsBySearchModel() -> receivedFromBackEnd -> size : {}", requireNonNull(productListResponseEntity.getBody()).getProducts().size());
        } else {
            searchService.resetSearchModel();
            productListResponseEntity = restTemplate.postForEntity(resourceUrl + "/searching/", searchService.getSearchModel(), ProductListAndMinMaxPrice.class);
            searchService.getSearchModel().setSearchFailure(true);
            searchService.getSearchModel().setSearchResultView(true);
        }
        return productListResponseEntity.getBody();
    }

Reply:
and to be total safe, you could null-check the entire body. Because if the BE fails, or the call is corrupted, you could break the productOverview page if list is empty, or the min - max price would be displayed as null. In that case, return a new ProductListAndMinMaxPrice object with default values.

Reply:
The whole body == null check is reverted to its original -> if the HttpStatus from the exchange is anything but ok it builds a default object to return.
The check to see if the return is 0 in size is moved to the backend, the backend sends a searchFailure = true in the response object now.
Setting this value in the frontEnd's searchModel is added in the successfull check (by new method for clarity), setting this is made safe with a Object.requireNonNull method.
All this was needed to not reset the model that would reset the price. AND the size = 0 of the search is more logical in the backEnd.

Reply:
agreed. As discussed, the requireNonNull throws a nullpointer if body is null. Which we don't want here.